helper/user_object.js

```diff
@@ -0,0 +1,107 @@
+// provides factory to create new users
+
+// sharedWith?: friend's uuid = boolean
+class UserDocumentMetadata {
+  constructor(isPublic, sharedWith) {
+    this.userDocObject = {
+      isPublic,
+      sharedWidth: Object.assign({}, sharedWith),
+    };
+  }
+
+  get getUserDocObject() {
+    return this.userDocObject;
+  }
+}
+
+class User {
+  constructor() {
+    this.user = {
+      uuid: [],
```

Comment:
uuid seems to be wrong 
refer to https://github.com/faraonc/hwsc-api-blocks/blob/master/lib/user.proto